File: rehovot-ads-packages/android/src/main/java/com/rehovottech/levelplay/ConsentManager.kt
package com.rehovottech.levelplay

// Privacy responsibilities live here so ad code remains focused on presentation.
class ConsentManager(
    private val logger: Logger,
) {
    private var coppaEnabled: Boolean = false

    public fun setConsent(options: ConsentRequest) {
        logger.d("TODO: Apply consent settings to Unity LevelPlay SDK.")
        logger.d("Consent options: $options")
    }

    public fun setCOPPA(enabled: Boolean) {
        coppaEnabled = enabled
        logger.d("TODO: Apply COPPA setting to Unity LevelPlay SDK.")
        logger.d("COPPA enabled: $coppaEnabled")
    }

    public fun setUserId(options: UserRequest) {
        logger.d("TODO: Apply user identity to Unity LevelPlay SDK.")
        logger.d("User options: $options")
    }

    public fun destroy() {
        logger.d("TODO: Reset consent and user identity state.")
        coppaEnabled = false
    }
}

File: rehovot-ads-packages/ios/Plugin/ConsentManager.swift
import Foundation

// Privacy management stays isolated from ad presentation logic.
final class ConsentManager {
    private let logger: Logger
    private var coppaEnabled: Bool = false

    init(logger: Logger) {
        self.logger = logger
    }

    func setConsent(_ options: ConsentRequest) {
        logger.debug("TODO: Apply consent settings to Unity LevelPlay SDK.")
        logger.debug("Consent options: \(options)")
    }

    func setCOPPA(_ enabled: Bool) {
        coppaEnabled = enabled
        logger.debug("TODO: Apply COPPA setting to Unity LevelPlay SDK.")
        logger.debug("COPPA enabled: \(coppaEnabled)")
    }

    func setUserId(_ options: UserRequest) {
        logger.debug("TODO: Apply user identity to Unity LevelPlay SDK.")
        logger.debug("User options: \(options)")
    }

    func destroy() {
        logger.debug("TODO: Reset consent and user identity state.")
        coppaEnabled = false
    }
}
